lib/github.ml

```diff
@@ -260,37 +222,24 @@ let curl_upload_archive ~token ~dry_run curl archive user repo release_id =
 
 let curl_open_pr ~token ~dry_run ~title ~distrib_user ~user ~branch ~body
     ~opam_repo curl =
-  let parse_url resp =
-    (* FIXME this is nuts. *)
-    let url =
-      Re.(
-        compile
-        @@ seq
-             [
-               bol;
-               str {|  "html_url":|};
-               rep space;
-               char '"';
-               group (rep (compl [ char '"' ]));
-             ])
-    in
-    let alread_exists = Re.(compile @@ str "A pull request already exists") in
-    try Ok (`Url Re.(Group.get (exec url resp) 1))
-    with Not_found ->
-      if Re.execp alread_exists resp then Ok `Already_exists
-      else R.error_msgf "Could not find html_url id in response:\n%s." resp
-  in
   let base, repo = opam_repo in
   let uri = strf "https://api.github.com/repos/%s/%s/pulls" base repo in
   let data =
-    strf {|{"title": %S,"base": "master", "body": %S, "head": "%s:%s"}|} title
-      body user branch
+    let open Yojson.Basic in
+    `Assoc
+      [
+        ("title", `String title);
+        ("base", `String "master");
+        ("body", `String body);
+        ("head", `String (strf "%s:%s" user branch));
+      ]
+    |> to_string
```

Comment:
I wonder if this change is really an improvement, wasn't it more readable before?

Reply:
Do you mean that the `strf` was more readable than the `Yojson` value?

I'm not sure it was but I'm quite used to `Yojson`'s notation. I think it's much better as now at least it's properly encoded as JSON so I'm definitely in favor of this change!

Reply:
Also in the future we could use `ppx_yojson`!

Reply:
I meant that it was shorter and closer to the json value produced.
Unfortunately ppx_yojson doesn't handle variables (for now) so we cannot use it here.